### ISSUE TITLE
Make necessary converting for OSM separate from core

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#939]
See e.g. the max speed parsing or weight/height parsing (#936) or conditional restrictions (#374)

Other related issues: #583

**Comment by boldtrn** [comment]
So, if I understand the problem in #936 correctly, the problem is that we cannot put it into the OSM module, since we use it in the core. Adding a dependency to the OSM module would create a circular dependency?

**Comment by karussell** [comment]
> Adding a dependency to the OSM module would create a circular dependency?

Yes. Just imagine a different input like from commercial data requires a different reader but should not require a different encoder.
